operator/src/test/java/org/keycloak/operator/testsuite/integration/TracingDeploymentTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator.testsuite.integration;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.Constants;
+import org.keycloak.operator.crds.v2alpha1.deployment.Keycloak;
+import org.keycloak.operator.crds.v2alpha1.deployment.ValueOrSecret;
+import org.keycloak.operator.crds.v2alpha1.deployment.spec.FeatureSpec;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_RESOURCE_ATTRIBUTES;
+import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_SERVICE_NAME;
+import static org.keycloak.operator.testsuite.utils.K8sUtils.deployKeycloak;
+
+@QuarkusTest
+public class TracingDeploymentTest extends BaseOperatorTest {
+
+    public static Keycloak getTestKeycloakDeployment(boolean disableProbes) {
+        var kc = BaseOperatorTest.getTestKeycloakDeployment(disableProbes);
+        kc.getSpec().setStartOptimized(false);
+        kc.getSpec().getAdditionalOptions().add(new ValueOrSecret("tracing-enabled", "true"));
+
+        var featureSpec = new FeatureSpec();
+        featureSpec.setEnabledFeatures(List.of("opentelemetry"));
+        kc.getSpec().setFeatureSpec(featureSpec);
+        return kc;
+    }
+
+    @Test
+    public void podNamePropagation() {
```

Comment:
Possibly we don't even need to move it to a separate test class, WDYT?

Reply:
I'd like to move here also the test case for Keycloak Tracing CR to have it separated. Does it work for you?

Reply:
Sure, if we move the other tracing test here, that seems good. Point is to keep them in a single place.

Reply:
Do you want move the other test here as well?

Reply:
I think it should be done in a separate PR to keep some sort of atomicity.

Reply:
Ok, works for me either way, don't want to block this PR on this.

My reasoning to suggest moving the other OTEL test (`testTracingSpec`) in this PR was that we're also moving `podNamePropagation` to a new test class here, so felt natural to me to move both at the same time. But it's not a blocker for me.